src/lib.rs

```diff
@@ -39,40 +39,26 @@ pub fn init() -> Result<()> {
 /// not available, it will return `Err(Error)`. If the specified date is not a holiday, it
 /// will return `Ok(None)`. Otherwise, it will return `Ok(Some(Holiday))`.
 pub fn get(country: Country, date: NaiveDate) -> Result<Option<Holiday>> {
-    let Some(data) = Lazy::get(&DATA) else {
-        return Err(Error::Uninitialized);
-    };
-
-    let data = data.read().map_err(|e| Error::LockError(e.to_string()))?;
-    let Some(map) = data.get(&country) else {
-        return Err(Error::CountryNotAvailable);
-    };
-
-    let Some(map) = map.get(&date.year()) else {
-        return Err(Error::YearNotAvailable);
-    };
-
-    Ok(map.get(&date).cloned())
+    get_map_for_country_any_year(country, date.year(), |map| map.get(&date).cloned())
 }
 
 /// Check if the specified date is a holiday. If the specified country or year is
 /// not available, it will return `Err(Error)`. If the date is not a holiday, it
 /// will return `Ok(false)`. Otherwise, it will return `Ok(true)`.
 pub fn contains(country: Country, date: NaiveDate) -> Result<bool> {
-    let Some(data) = Lazy::get(&DATA) else {
-        return Err(Error::Uninitialized);
-    };
+    get_map_for_country_any_year(country, date.year(), |map| map.get(&date).is_some())
+}
 
+fn get_map_for_country_any_year<T>(
+    country: Country,
+    year: Year,
+    result_getter: impl FnOnce(&BTreeMap<NaiveDate, Holiday>) -> T,
+) -> Result<T> {
```

Comment:
Could you simply do like
```suggestion
fn get_map_for_country_any_year(
    country: Country,
    year: Year,
) -> Result<&'static BTreeMap<NaiveDate, Holiday>> {
```
?

Reply:
I tried this as well but this gives the following error:
```rs
error[E0515]: cannot return value referencing local variable `data`
  --> src/lib.rs:62:5
   |
61 |     let map = data.get(&country).ok_or(Error::CountryNotAvailable)?;
   |               ---- `data` is borrowed here
62 |     map.get(&year).ok_or(Error::YearNotAvailable)
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ returns a value referencing data owned by the current function
```

I think the `RwLockReadGuard` doesn't have a static lifetime because it lives in the current function.

Reply:
I split this into two separate functions to get around this issue (`get_holiday_map` and `get_map_for_country_and_year`)